src/Entity/BillThirdParty.php

```diff
@@ -63,4 +67,32 @@ public function setAccountNumber($accountNumber)
 
         return $this;
     }
+
+    /**
+     * @param DOMDocument|null $document
+     * @return DOMElement
+     */
+    public function toNode(DOMDocument $document = null)
+    {
+        if (null === $document) {
+            $document = new DOMDocument();
+        }
+
+        $node = $document->createElement('BillThirdParty');
+
+        $btpNode = $node->appendChild($document->createElement('BillThirdPartyShipper'));
+        $btpNode->appendChild($document->createElement('AccountNumber', $this->getAccountNumber()));
```

Comment:
Can we offload in here also to custom entities? So a 'BillThirdPartyShipper' with it's own `toNode` method? It feels more like following the existing flow than creating sub entities inside this class.